src/query/storages/parquet/src/parquet_rs/source.rs

```diff
@@ -325,6 +344,36 @@ impl ParquetSource {
                 part.file.as_str(),
             )?;
         }
+        // The schema of the table in iceberg may be inconsistent with the schema in parquet
+        let reader = if matches!(self.source_type, ParquetSourceType::Iceberg)
```

Comment:
no need to check `matches!(self.source_type, ParquetSourceType::Iceberg)`.

we also need it in StageTable, eg: `select * from @test`